File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.py
def solution(phone_book):
    N = len(phone_book)
    
    phone_book.sort(key=lambda x:len(x))
    set_ = set()
    for i in range(N):
        tmp = ""
        for j in range(len(phone_book[i])):
            tmp += phone_book[i][j]
            if tmp in set_: return False
        set_.add(phone_book[i])
    return True

File: 프로그래머스/2/42577. 전화번호 목록/전화번호 목록.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <iostream>
#include <set>

using namespace std;

bool solution(vector<string> phone_book) {
    
    set<string> set_;
    for (int i = 0; i < phone_book.size(); ++i)
    {
        set_.insert(phone_book[i]);
    }
    
    for (int i = 0; i < phone_book.size(); ++i)
    {
        string number = "";
        for (int j = 0; j < phone_book[i].size(); ++j)
        {
            number += phone_book[i][j];
            if ((set_.find(number) != set_.end()) && (phone_book[i] != number))
            {
                return false;
            }
        }
    }
    
    return true;
}